test/e2e/kubernetes/resource.go

```diff
@@ -0,0 +1,52 @@
+package kubernetes
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	resourceWaitTimeout  = 3 * time.Minute
+	resourcePollInterval = 5 * time.Second
+	maxConsecutiveErrors = 3
+)
+
+// WaitForResource is a generic function to wait until a kubernetes resource is ready.
+func WaitForResource(
```

Comment:
I appreciate the intention here
However, are we gaining enough here to justify the complexity? It seems like the only reusable piece is the "error retry". But In order to do that, we need to define a pretty complex interface that takes 2 closures.

I wonder if maybe we should focus on that: have a reusable piece that retries errors for a determined amount of times. Then if you want to wait for resource ready, you wrap you code with a wait.Poll, which is already available everywhere.